src/NuGet.Services.AzureSearch/DownloadTransferrer.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.AzureSearch.AuxiliaryFiles;
+
+namespace NuGet.Services.AzureSearch
+{
+    public class DownloadTransferrer : IDownloadTransferrer
+    {
+        private readonly IAuxiliaryFileClient _auxiliaryFileClient;
+        private readonly IDatabaseAuxiliaryDataFetcher _databaseFetcher;
+        private readonly ILogger<DownloadTransferrer> _logger;
+
+        public DownloadTransferrer(
+            IAuxiliaryFileClient auxiliaryFileClient,
+            IDatabaseAuxiliaryDataFetcher databaseFetcher,
+            ILogger<DownloadTransferrer> logger)
+        {
+            _auxiliaryFileClient = auxiliaryFileClient ?? throw new ArgumentException(nameof(auxiliaryFileClient));
+            _databaseFetcher = databaseFetcher ?? throw new ArgumentNullException(nameof(databaseFetcher));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task<DownloadTransferResult> GetTransferChangesAsync(DownloadData downloads)
```

Comment:
What do you think about the name `InitializeTransferChangesAsync`? Seems like this should only be called in a case like Db2AzureSearch

Reply:
I renamed this to `InitializeDownloadTransfers` and renamed the other method to `UpdateDownloadTransfers`